pkg/util/file.go

```diff
@@ -37,3 +38,28 @@ func ReadFile(path string) (string, error) {
 
 	return "", errors.New("Timeout while trying to read the file")
 }
+
+// AtomicWrite firsts writes data to a temp file, then renames to the destination file.
+// This ensures that the destination file is never partially written.
+func AtomicWrite(fileName string, data []byte, perm os.FileMode) error {
+	f, err := os.CreateTemp(filepath.Dir(fileName), filepath.Base(fileName)+".tmp")
+	if err != nil {
+		return err
+	}
+	tmpName := f.Name()
```

Comment:
If we take any of the error paths here, we'll leak the temp file. I think we should defer a delete - if the final rename succeeds, the delete will be a no-op as the temp file won't exist.
```suggestion
	tmpName := f.Name()
	defer os.Remove(tmpName)
```